File: WEEK 1/JavaScript/Hard Drive Storage Capacity Conversion.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function getUserInput(prompt) {
  return new Promise((resolve) => {
    rl.question(prompt, (input) => {
      resolve(parseFloat(input));
    });
  });
}

async function main() {
  try {
    const manufacturerGB = await getUserInput('Enter the hard drive size as specified by the manufacturer (in GB): ');

    if (manufacturerGB < 0) {
      console.log('Please enter a non-negative value for hard drive size.');
      return;
    }

    const bytesInGBManufacturer = 1000 * 1000 * 1000; // 1 GB (manufacturer) = 1,000,000,000 bytes
    const bytesInGBStandard = 1024 * 1024 * 1024; // 1 GB (standard) = 1,073,741,824 bytes

    const actualGB = manufacturerGB * (bytesInGBManufacturer / bytesInGBStandard);

    console.log(`The actual storage capacity of the hard drive is approximately ${actualGB.toFixed(2)} GB.`);
  } catch (error) {
    console.error(error);
  } finally {
    rl.close();
  }
}

main();

File: WEEK 1/java/Hard Drive Storage Capacity Conversion.java
import java.util.Scanner;

public class HardDriveCapacityConverter {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        // Prompt the user to enter the hard drive size as specified by the manufacturer (in GB)
        System.out.print("Enter the hard drive size as specified by the manufacturer (in GB): ");
        
        if (input.hasNextDouble()) {
            double manufacturerSizeGB = input.nextDouble();

            // Calculate the actual storage capacity (in GB) based on the computer memory standard
            double actualSizeGB = manufacturerSizeGB * 1000 * 1000 * 1000 / (1024 * 1024 * 1024);

            // Output the actual storage capacity
            System.out.println("Actual storage capacity: " + actualSizeGB + " GB");
        } else {
            System.out.println("Invalid input. Please enter a valid number for the hard drive size.");
        }

        input.close();
    }
}
